lib/metrics/client-metrics.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+/**
+ * Represents a base class that is used to measure events from the server and the client as seen by the driver.
+ * @alias module:metrics~ClientMetrics
+ * @interface
+ */
+class ClientMetrics {
```

Comment:
Just had a realization.  Currently, all metrics are based on some kind of activity, what about state-based metrics?  For example both the python and java driver expose inflight request counters, total number of connections, connected host count, etc.  Technically those could be event based too, but wonder if sharing come kind of state as a metric is something we should consider.

Reply:
State info is currently exposed via `client.getState()`. I was also wondering whether we should expose it in the metrics but I didn't see it clearly (should this be refreshed on each call / periodically ?) and the user can always implement a provider specific gauge using `ClientState` information.

Reply:
> the user can always implement a provider specific gauge using ClientState information.

Just realized that's precisely what I'm doing already :man_facepalming: .  I think I agree, lets keep `ClientMetrics` to only activity-based things, and users can use `Client.getState` to get client & host state.